context/context.go

```diff
@@ -133,6 +148,10 @@ func (c *ctx) Reset() {
 			// Free any large arrays that are created
 			c.dep.closers = nil
 		} else {
+			for i := range c.dep.closers {
```

Comment:
Unfortunately the array that is backing the slice still holds onto this memory:
http://stackoverflow.com/a/16972044